monitor/flowgen/bridging/bridging.go

```diff
@@ -96,48 +119,68 @@ func openOrder(
 			Amount: depositWithFee.Amount,
 		},
 		Expenses: []solvernet.Expense{}, // Explicit expense not required for native transfer calls.
-		Calls: []bindings.SolverNetCall{
-			{
-				Target: owner,
-				Value:  expense.Amount,
-			},
-		},
+		Calls:    []bindings.SolverNetCall{call.ToBinding()},
 	}
 
 	orderID, err := solvernet.OpenOrder(ctx, networkID, conf.srcChain, backends, owner, orderData)
 	if err != nil {
 		return types.Result{}, false, errors.Wrap(err, "open order")
 	}
 
+	req := stypes.CheckRequest{
+		SourceChainID:      conf.srcChain,
+		DestinationChainID: conf.dstChain,
+		FillDeadline:       uint32(time.Now().Add(time.Hour).Unix()), //nolint:gosec // unix timestamp won't overflow
+		Deposit:            stypes.AddrAmt(orderData.Deposit),
+		Expenses: []stypes.Expense{{
+			Amount: call.Value,
+		}},
+		Calls: []stypes.Call{stypes.Call(call)},
+	}
+	resp, err := solverClient.Check(ctx, req)
+	if err != nil {
+		return types.Result{}, false, errors.Wrap(err, "check solving")
+	}
+
+	if resp.Rejected {
```

Comment:
I suggest sticking to PR description which states: "only skip on insufficient inventory", that implies erroring on others which are unexpected.